standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -273,6 +273,7 @@ public CompactionInfo findNextToCompact(FindNextCompactRequest rqst) throws Meta
           info.type = TxnUtils.dbCompactionType2ThriftType(rs.getString(5).charAt(0));
           info.poolName = rs.getString(6);
           info.properties = rs.getString(7);
+          info.workerId = rqst.getWorkerId();
```

Comment:
is this needed?

Reply:
Yes, this is required due to moving MrCompactor behind CompactorFactory. See MrCompactor:
` public void run(HiveConf conf, Table t, StorageDescriptor sd, ValidWriteIdList writeIds,
           CompactionInfo ci, IMetaStoreClient msc, AcidDirectory dir) throws IOException {
    String jobName = ci.workerId + "-compactor-" + ci.getFullPartitionName();`

Reply:
👍